### PR TITLE
Commands

### PR DESCRIPTION
**Pull request by achille-roussel**
@f2prateek 
@tejasmanohar 
@andreiko 

Hey guys, I added support for commands as first argument of the program (like in `git checkout ...`, "checkout" is the command).

Please take a look and let me know if you'd like to see anything changed.

### CODE REVIEW

File: load.go [hunk 1]
```diff
@@ -51,13 +53,20 @@ func LoadWith(cfg interface{}, ld Loader) (args []string) {
 	return
 }
 
+// A Command represents a command supported by a configuration loader.
+type Command struct {
+	Name string // name of the command
+	Help string // help message describing what the command does
+}
+
 // A Loader can be used to provide a costomized configurable for loading a
```
Comment (f2prateek) [inline load.go]: Unrelated to this change but should be `customized`.

Comment (achille-roussel) [inline load.go]: 👍

**Comment by achille-roussel** [comment]
@f2prateek si that good to go?

**Comment by f2prateek** [comment]
LGTM!

**Comment by achille-roussel** [comment]
Thanks :)

**Pull request merged**